libyul/optimiser/StructuralSimplifier.cpp

```diff
@@ -45,6 +45,26 @@ void StructuralSimplifier::operator()(Block& _block)
 	popScope();
 }
 
+boost::optional<dev::u256> StructuralSimplifier::hasLiteralValue(Expression const& _expression) const
+{
+	Expression const* expr = &_expression;
+
+	if (expr->type() == typeid(Identifier))
+	{
+		Identifier const& ident = boost::get<Identifier>(*expr);
+		if (m_value.count(ident.name))
+			expr = m_value.at(ident.name);
+	}
+
```

Comment:
Please either handle the case or assert that `expr` is not `nullptr` here.

Reply:
But I am explicitly checking that it can't be null in line 55? I can add another assert, but from what I see that case is handled/excluded already.

Reply:
You are only checking that the key is present in `m_value`, the value can still be a nullptr. This is currently not the case in DataFlowAnalyzer, but that chan change any time.